simplemma/lemmatizer.py

```diff
@@ -233,8 +204,12 @@ def _return_lemma(
     if len(token) <= limit:
         return candidate
     # subword decomposition: predefined prefixes (absent from vocabulary if they are not words)
-    if candidate is None:
-        candidate = _prefix_search(token, lang, datadict)  # type: ignore[arg-type]
+    if candidate is None and lang in PREFIXES:
+        prefix = PREFIXES[lang](token)
```

Comment:
These lines could be listed in a separate function since we're re-organizing the code